Controllers/BookStoreItemController.cs

```diff
@@ -0,0 +1,187 @@
+﻿using System;
+namespace BookStore.Controllers
+{
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using BookStore.Models;
+using AutoMapper;
+using BookStore.Interfaces;
+using BookStore.DTO;
+    using BookStore.Services;
+    using BookStore.Enums;
+
+    public class BookStoreItemController : Controller
+    {
+        private readonly IBookStoreService _bookStoreService;
+
+        private readonly IMapper _mapper;
+
+        public BookStoreItemController(IBookStoreService bookStoreService, IMapper mapper)
+        {
+            _bookStoreService = bookStoreService;
+            _mapper = mapper;
+        }
+
+        // GET: BookStoreItem
+        public IActionResult Index(string searchString)
+        {
+            if (_bookStoreService.Get() == null)
+            {
+                return Problem("Stores list is empty.");
+            }
+
+            var stores = from s in _bookStoreService.BookStoreItems()
+                           select s;
+
+            if (!String.IsNullOrEmpty(searchString))
+            {
+                stores = stores.Where(s => s.Product.Name!.Contains(searchString));
+            }
+            return View(stores.ToList());
+        }
+
+        // GET: BookStoreItem/Details/5
+        public async Task<IActionResult> Details(int id)
+        {
+            if (id == null || _bookStoreService.Get(id) == null)
+            {
+                return NotFound();
+            }
+
+            var bookStoreItem = _bookStoreService.Get(id);
+
+            return View(bookStoreItem);
+        }
+
+        // GET: BookStoreItem/Create
+        public IActionResult Create()
+        {
+            ViewData["Name"] = new SelectList(_bookStoreService.Products(), "Name", "Name");
+            return View();
+        }
+
+        // POST: BookStoreItem/Create
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public IActionResult Create(string Product, [Bind("Available,Booked,Sold")] BookStoreItemDto bookStoreItem)
+        {
+            var bookedCount = _bookStoreService.Products()
+                .Where(b => b.Name == Product)
+                .FirstOrDefault().Bookings
+                .Where(s => s.StatusId == (int)Statuses.APPROVED || s.StatusId == (int)Statuses.SUBMITED).Count();
+
```

Comment:
Such an approach will lead to performance issues: the method _bookStoreService.Products() joints products to the booking, statuses, etc => and fetches it to the memory => after that you are filtering it by your needs. Use DB logic for filtering items set and after that fetch it to the memory. Look into IQueryable